submissions/mustbefail/oop/index.js

```diff
@@ -0,0 +1,86 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+  Complete the below for code reviewers' convenience:
+
+  Code repository: _put repo URL here_
+  Web app: _put project's github pages URL here_
+  */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+class Inhabitant {
+  constructor(name, gender, phrase, friends, legs) {
+    this.friends = friends ? friends : ['no friends'];
+    this.name = name;
+    this.gender = gender;
+    this.legs = legs;
+    this.phrase = phrase;
+  }
+  saying() {
+    return this.phrase;
+  }
+
+  toString() {
+    return `Hello! I am <strong>${this.species}</strong>, my name is <strong>${
+      this.name
+    }</strong>, my gender is <strong>${this.gender}</strong>, I have <strong>${
+      this.hands
+    }</strong> arms and <strong>${
+      this.legs
+    }</strong> legs and you know what: <strong>${this.saying()}</strong>. My friends: <strong>${
+      this.friends
+    }</strong>`;
+  }
+}
+
+class Cat extends Inhabitant {
+  constructor(name, gender, phrase, friends, legs = 4) {
+    super(name, gender, phrase, friends, legs);
+    this.species = 'cat';
+  }
+}
+
+class Dog extends Inhabitant {
+  constructor(name, gender, phrase, friends, legs = 4) {
+    super(name, gender, phrase, friends, legs);
+    this.species = 'dog';
```

Comment:
Isn't species a shared property worth defining in super class?
You had a nice solution for this before. Not every comment is about fixing something :)

Reply:
> This way we do not ask a developer to specify species when constructing a cat and the species is hardcoded.
> It's a good approach.

Misunderstood